codec.go

```diff
@@ -0,0 +1,173 @@
+package codec
+
+import (
+	"bufio"
+	"errors"
+	"io"
+	"math"
+
+	"zikichombo.org/sound"
+	"zikichombo.org/sound/ops"
+	"zikichombo.org/sound/sample"
+)
+
+// ErrUnknownCodec is an error representing
+var ErrUnknownCodec = errors.New("unknown codec")
+
+// ErrUnsupportedSampleCodec can be used by codec implementations
+// which receive a request for encoding or decoding with a sample codec
+// that is unsupported or doesn't make sense.
+var ErrUnsupportedSampleCodec = errors.New("unsupported sample codec")
+
+// AnySampleCodec is a sample.Codec which is used as a wildcard sample.Codec
+// in this interface.  Codecs which are not based on PCM data, such as aac/mp3/ogg/vorbis
+//
+var AnySampleCodec = sample.Codec(-1)
```

Comment:
Perhaps look into having the zero value be the *any sample codec*. That way, the zero value is usable by default, and if users care, they can specify a specific codec in `Codec.DefaultSampleCodec`.

If so, the sample package would be updated,

```diff
 const (
+	SAny Codec = iota
-	SInt8 Codec = iota
+	SInt8
 	SByte
 	SInt16L   // little endian 2 byte
 	SInt16B   // big endian 2 byte
 	SInt24L   // little endian 3 byte
 	SInt24B   // big endian 3 byte
 	SInt32L   // little endian 4 byte
 	SInt32B   // big endian 4 byte
 	SFloat32L // float32 bits stored uint32 little endian
 	SFloat32B // float32 bits stored uint32 big endian
 	SFloat64L // float64 bits stored uint64 little endian
 	SFloat64B // float64 bits stored uint64 big endian
 )
```

On a related note, how does `SInt8` and `SByte` differ?